include/swift/AST/Expr.h

```diff
@@ -4687,6 +4699,53 @@ class ApplyExpr : public Expr {
     Bits.ApplyExpr.NoAsync = noAsync;
   }
 
+  const DeclContext *getDeclContext() const {
+    assert(declContext
+           && "declContext must be initialized with setDeclContext");
+    return declContext;
+  }
+  void setDeclContext(const DeclContext *ctxt) {
+    declContext = ctxt;
+  }
+
+  bool makesUnsatisfiedIsolationJump() const {
+    return Bits.ApplyExpr.MakesUnsatisfiedIsolationJump;
+  }
+
+  // Check whether this apply makes an unsatisfied isolation jump from an
+  // actor-isolated to nonisolated context
+  bool makesUnsatisfiedIsolationExit() const {
+    return makesUnsatisfiedIsolationJump() &&
+           !unsatisfiedIsolationJumpCallee.isActorIsolated();
+  }
```

Comment:
I'm surprised this isn't a callee != caller isolation check. What if both are isolated but to different actors?